### PR TITLE
More RPC changes for SP

### PR DESCRIPTION
**Pull request by zathras-crypto**

### CODE REVIEW

File: src/mastercore.cpp [hunk 1]
```diff
@@ -3356,9 +3356,8 @@ if (fHelp || params.size() != 4)
   if (NULL == property) // property ID does not exist
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Property ID does not exist");
 
-  // ***This test also disabled because getDivisible still needs to be implemented
-  // bool divisible = property.getDivisible();
-  bool divisible = true; // hard code to true temporarily
+  bool divisible = false;
+  divisible=getSP(propertyId)->isDivisible();
```
Comment (m21) [inline src/mastercore.cpp:3360]: NULL pointer check is needed.........
below too, similar code

File: src/mastercore.cpp [hunk 2]
```diff
@@ -3995,31 +3993,39 @@ Value getallbalancesforaddress_MP(const Array& params, bool fHelp)
...
-        else
-        {
-        propertybal.push_back(Pair("balance", getMPbalance(address, propertyId, MONEY)));
-        propertybal.push_back(Pair("reservedbyoffer", getMPbalance(address, propertyId, SELLOFFER_RESERVE)));
-        if (propertyId<3) propertybal.push_back(Pair("reservedbyaccept", getMPbalance(address, propertyId, ACCEPT_RESERVE)));
-        }
-        response.push_back(propertybal);
-//    }
-return response;
+    addressTally->init();
```
Comment (m21) [inline src/mastercore.cpp:4001]: NULL pointer check

File: src/mastercore.cpp [hunk 3]
```diff
@@ -3417,9 +3416,8 @@ Value getbalance_MP(const Array& params, bool fHelp)
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Property ID does not exist");
 #endif
 
-    // ***This test also disabled because getDivisible still needs to be implemented
-    // bool divisible = property.getDivisible();
-    bool divisible = true; // hard code to true temporarily
+    bool divisible = false;
+    divisible=getSP(propertyId)->isDivisible();
```
Comment (m21) [inline src/mastercore.cpp:3420]: NULL pointer check

File: src/mastercore.cpp (hunk 2 above)
Comment (zathras-crypto) [inline src/mastercore.cpp:4001]: right above it
 if (NULL == addressTally) // property ID does not exist
          throw JSONRPCError(RPC_INVALID_PARAMETER, "Address not found");

Comment (zathras-crypto) [inline src/mastercore.cpp:4001]: Badly commented tho ha

File: src/mastercore.cpp (hunk 1 above)
Comment (zathras-crypto) [inline src/mastercore.cpp:3360]: Yeah - there is a null check above again
  if (NULL == property) // property ID does not exist
              throw JSONRPCError(RPC_INVALID_PARAMETER, "Property ID does not exist");
but I should just be reusing that property object - will amend

File: src/mastercore.cpp (hunk 3 above)
Comment (zathras-crypto) [inline src/mastercore.cpp:3420]: same as above - will amend

**Pull request merged**